zookeeper-server/src/main/java/org/apache/zookeeper/Op.java

```diff
@@ -253,6 +254,142 @@ void validate() throws KeeperException {
         PathUtils.validatePath(path);
     }
 
+    /**
+     * "Fluent builder" for {@link Create} operations.
+     */
+    public static class CreateBuilder {
+        private String path;
+        private byte[] data;
+        private List<ACL> acl;
+        private Integer createModeFlag;
+        private CreateMode createMode;
+        private boolean returnStat = false;
+        private Long ttl;
+
+        /**
+         * Sets the path for the node.  Must be set before calling
+         * {@link #build()}.
+         *
+         * @param path  the path for the node
+         * @return the builder, for chaining.
+         */
+        public CreateBuilder setPath(String path) {
+            this.path = path;
+            return this;
+        }
+
+        /**
+         * Sets the initial data for the node, or {@code null}.
+         *
+         * @param data  the initial data for the node
+         * @return the builder, for chaining.
+         */
+        @SuppressFBWarnings({"EI_EXPOSE_REP", "EI_EXPOSE_REP2"})
+        public CreateBuilder setData(byte[] data) {
+            this.data = data;
+            return this;
+        }
+
+        /**
+         * Sets the acl for the node.  Must be set before calling
+         * {@link #build()}.
+         *
+         * @param acl  the acl for the node
+         * @return the builder, for chaining.
+         */
+        public CreateBuilder setACL(List<ACL> acl) {
+            this.acl = acl;
+            return this;
+        }
+
+        /**
+         * Specifyies whether the node to be created is ephemeral,
+         * sequential, etc. using the integer encoding.  Either this
+         * or {@link #setCreateMode} must be used before calling
+         * {@link #build()}.
+         *
+         * @param createModeFlag  the create mode encoded as an integer
+         * @return the builder, for chaining.
+         */
+        public CreateBuilder setCreateModeFlag(Integer createModeFlag) {
```

Comment:
why "Integer" ?
is it expected to pass "null" ?

I am fine with having `Integer` as internal value, but maybe here in this point we should have only `int`

Reply:
Yes, the intent was to allow (re)setting these properties to `null` (which is their default value), otherwise there is no way to put the object back into its default state.

(As I suppose you have noticed, `null` is notably used as a marker for `createMode` and `createModeFlag`, which are exclusive.)

But I can switch it to `int`; I don't expect that many people will want to reuse `CreateBuilder` instances.

Reply:
Done.